File: ProductStoreAPI/ProductStoreAPI.Models/Business/Product/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ProductStoreAPI.Models.Business.Product
{
	public sealed class Product
	{
		[Key]
		[Column("id")]
		public long ProductId { get; init; }

		[Required]
		[Column("name")]
		public required string Name { get; init; }

		[Column("description")]
		public string Description { get; init; }

		[Column("price")]
		public decimal Price { get; init; } = decimal.Zero;

		[Column("stock_quantity")]
		public int StockQuantity { get; init; } = 0;

		[Column("category")]
		public string Category { get; init; }

		[Column("image_url")]
		public string ImageUrl { get; init; }

		[Column("manufacturer")]
		public string Manufacturer { get; init; }

		[Column("is_active")]
		public bool IsActive { get; init; } = false;
	}
}

File: ProductStoreUI/src/app/modules/management/pages/product/models/Product.ts
export class Product {
	productId: number;
	name: string;
	description: string;
	price: number;
	stockQuantity: number;
	category: string;
	imageUrl: string;
	manufacturer: string;
	isActive: boolean;

	constructor(
		productId: number,
		name: string = '',
		description: string = '',
		price: number = 0,
		stockQuantity: number = 0,
		category: string = '',
		imageUrl: string = '',
		manufacturer: string = '',
		isActive: boolean = false,
	) {
		this.productId = productId;
		this.name = name;
		this.description = description;
		this.price = price;
		this.stockQuantity = stockQuantity;
		this.category = category;
		this.imageUrl = imageUrl;
		this.manufacturer = manufacturer;
		this.isActive = isActive;
	}
}